### DISCUSSION TITLE
Open IBM i source in browse mode instead of edit mode

### DISCUSSION THREAD

**Discussion by JoeCharlesMorgan** [codefori/vscode-ibmi#1412] [Q&A] [answered]
Is this possible?  Am not able to find a VS json setting or thread on this website for this.

**Comment by sebjulliand** [comment]
Hi @JoeCharlesMorgan ,
for now it is possible to `protect` a filter - from the filter settings:
![image](https://github.com/halcyon-tech/vscode-ibmi/assets/11096890/849f55d4-3208-4b82-bbdd-d7da8ece8a13)

This will open every member from this filter in read only mode.

Maybe adding a "Browse" action in members' right click menu could be nice too. I'll keep that in mind.

**Reply by sebjulliand** [reply, marked as answer]
If this gets approved and merge, there will be a Browse action for stream files and members: https://github.com/halcyon-tech/vscode-ibmi/pull/1413

**Comment by JoeCharlesMorgan** [comment]
Sebastien,
Thanks for the update.  One more question about this.  I open a lot of source with CTRL- ALT- P.  Is there also a setting somewhere to only open in browse mode this way?

Thanks!
Joe

On Jul 6, 2023, at 4:16 AM, Sébastien Julliand ***@***.***> wrote:

﻿

If this gets approved and merge, there will be a Browse action for stream files and members: #1413<https://github.com/halcyon-tech/vscode-ibmi/pull/1413>

**Reply by sebjulliand** [reply]
Yes! You can use `CTRL+ALT+O` to open source in read only mode.

**Comment by JoeCharlesMorgan** [comment]
Thanks again!!!

On Jul 6, 2023, at 9:07 AM, Sébastien Julliand ***@***.***> wrote:

﻿

Yes! You can use CTRL+ALT+O to open source in read only mode.

**Reply by sebjulliand** [reply]
Anytime 😄
